src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs

```diff
@@ -0,0 +1,243 @@
+            "The supported properties for directory are: CacheControl, ContentDisposition, ContentEncoding, ContentLanguage.", 
+            Mandatory = false)]
+        public Hashtable Property
+        {
+            get
+            {
+                return BlobProperties;
+            }
+
+            set
+            {
+                BlobProperties = value;
+            }
+        }
+        private Hashtable BlobProperties = null;
+
+        [Parameter(HelpMessage = "Specifies metadata for the directory or file.", 
+            Mandatory = false)]
+        public Hashtable Metadata
+        {
+            get
+            {
+                return BlobMetadata;
+            }
+
+            set
+            {
+                BlobMetadata = value;
+            }
+        }
+        private Hashtable BlobMetadata = null;
+
+        [Parameter(HelpMessage = "Sets POSIX access control rights on files and directories. Create this object with New-AzDataLakeGen2ItemAclObject.", Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public PSPathAccessControlEntry[] Acl { get; set; }
+
+        // Overwrite the useless parameter
+        public override int? ConcurrentTaskCount { get; set; }
+
+
+        /// <summary>
+        /// Initializes a new instance of the SetAzDataLakeGen2ItemCommand class.
+        /// </summary>
+        public SetAzDataLakeGen2ItemCommand()
+            : this(null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the SetAzDataLakeGen2ItemCommand class.
+        /// </summary>
+        /// <param name="channel">IStorageBlobManagement channel</param>
+        public SetAzDataLakeGen2ItemCommand(IStorageBlobManagement channel)
+        {
+            Channel = channel;
+        }
+        
+        /// <summary>
+        /// execute command
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            IStorageBlobManagement localChannel = Channel;
+            BlobRequestOptions requestOptions = RequestOptions;
+
+            bool foundAFolder = false;
+            CloudBlockBlob blob = null;
+            CloudBlobDirectory blobDir = null;
+            if (ParameterSetName == ManualParameterSet)
+            {
+                CloudBlobContainer container = GetCloudBlobContainerByName(localChannel, this.FileSystem).ConfigureAwait(false).GetAwaiter().GetResult();
+                foundAFolder = GetExistDataLakeGen2Item(container, this.Path, out blob, out blobDir);
+            }
+            else //BlobParameterSet
+            {
+                if (!InputObject.IsDirectory)
+                {
+                    blob = (CloudBlockBlob)InputObject.ICloudBlob;
+                }
+                else
+                {
+                    blobDir = InputObject.CloudBlobDirectory;
+                    foundAFolder = true;
+                }
+            }
```

Comment:
Seems this block appear many times. Can this be refacted into a function?

Reply:
I don’t prefer to make it into a function. since it depend on parameter set name, which is defined in each cmdlets. And since these cmdlet share the same base class, I don’t want to add the parameter set name to base class.

Reply:
OK, there isn't a better to do this.